api/admin.go

```diff
@@ -107,7 +107,15 @@ func saveConfig(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	err := app.SaveConfig(cfg)
+	cfg.SetDefaults()
+
+	err := cfg.IsValid()
+	if err != nil {
+		c.Err = err
+		return
+	}
```

Comment:
Actually it looks like `app.SaveConfig()` already sets defaults and checks for validity, so I don't think we need to add those here